pkg/operator/controller/ingress/networkpolicy.go

```diff
@@ -0,0 +1,121 @@
+package ingress
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+	controller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	networkingv1 "k8s.io/api/networking/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ensureRouterNetworkPolicy ensures the per-ingresscontroller NetworkPolicy that
+// allows ingress to router pods and egress to the network exists and is up to date.
+func (r *reconciler) ensureRouterNetworkPolicy(ic *operatorv1.IngressController) error {
```

Comment:
Any reason this `ensure` function doesn't return the object or Boolean value?  It doesn't strictly need to do so, so I don't object if you prefer to keep it simple, but it deviates from the established pattern for `ensure` methods.